llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -1764,6 +1764,22 @@ Instruction *InstCombinerImpl::visitFDiv(BinaryOperator &I) {
     return replaceInstUsesWith(I, Pow);
   }
 
+  // powi(X, Y) / X --> powi(X, Y-1)
+  // This is legal when (Y - 1) can't wraparound, in which case reassoc and nnan
+  // are required.
+  // Todo: Multi-use may be also better off creating Powi(x,y-1)
+  if (I.hasAllowReassoc() && I.hasNoNaNs() &&
+      match(Op0, m_OneUse(m_Intrinsic<Intrinsic::powi>(m_Specific(Op1),
+                                                       m_Value(Y)))) &&
+      computeOverflowForSignedSub(Y, ConstantInt::getAllOnesValue(Y->getType()),
```

Comment:
```suggestion
      computeOverflowForSignedSub(Y, ConstantInt::get(Y->getType(), 1),
```
Apologize for my mistake.

Reply:
Apply your comment, thanks @dtcxzyw for your carefully check.